drracket/drracket/private/tools.rkt

```diff
@@ -529,68 +527,56 @@
 
      (define (populate-listing!)
        (send listing clear)
-       (for-each
-        (lambda (entry+it)
-          (send listing append
-                (car entry+it)
-                (cdr entry+it)))
-        (sort (map (lambda (it) (cons (tool-list-entry it) it))
-                   installed-tools)
-              (lambda (a b)
-                (string<? (car a) (car b))))))
+       (for ([entry+it (in-list (sort (map (lambda (it) (cons (tool-list-entry it) it))
+                                           installed-tools)
+                                      (lambda (a b) (string<? (car a) (car b)))))])
```

Comment:
This should be different; possibly not in the purview of these changes?

Reply:
Hmm. I don't know of a good way to make this code much better, since `sort` needs to transform the whole list. That `sort` expression can be made a _little_ simpler though since instead of passing that lambda it could pass `(sort ... string<? #:key car)`. I already have a rule for that - I'm assuming it didn't fire here because the change hit the fix limit.

Reply:
As an aside, that code is _much_ easier to write with transducers:

```scheme
(transduce installed-tools
           (indexing tool-list-entry)
           (sorting string<=> #:key entry-key)
           #:into
           (into-for-each
             (lambda (entry+it)
               (send listing append (car entry+it) (cdr entry+it)))))
```

Reply:
Oh, I just meant the simple change using `sort`'s `#:key` argument. Sorry for being so opaque! Glad to hear that it'll get fixed on its own eventually! :)

The transducer code doesn't look much easier to me, but probably I'm missing something?

Reply:
Oh gotcha, yeah there's a rule for that `sort` fix. I thought you were talking about how indented the whole `sort` expression got because of it being on the right-hand-side of a `for` clause.

For the transducer code, I ought to have been clearer. The code I gave is equivalent to the whole `for-each` expression and `for` loop. These three snippets are all equivalent:

```scheme
(for-each
 (lambda (entry+it)
   (send listing append
         (car entry+it)
         (cdr entry+it)))
 (sort (map (lambda (it) (cons (tool-list-entry it) it))
            installed-tools)
       (lambda (a b)
         (string<? (car a) (car b)))))
```

```scheme
(for ([entry+it (in-list (sort (map (lambda (it) (cons (tool-list-entry it) it))
                                    installed-tools)
                               (lambda (a b) (string<? (car a) (car b)))))])
  (send listing append (car entry+it) (cdr entry+it)))
```

```scheme
(transduce installed-tools
           (indexing tool-list-entry)
           (sorting string<=> #:key entry-key)
           #:into
           (into-for-each
            (lambda (entry+it)
              (send listing append (car entry+it) (cdr entry+it)))))
```

It's pretty subjective (which I did not make clear in my previous comment) but I find the transducer version much easier to read assuming you know how to use `transduce`, since it's just a top-to-bottom series of steps applied to the initial list. Whenever I need to sort data and also do some transformation before or after, I usually use transducers because it's pretty hard to work `sort` into a `for` loop nicely. Plus `sort` is frustrating to use on non-lists.